src/middleware/controller/paginationUser-controller.js

```diff
@@ -0,0 +1,64 @@
+exports.paginateUser = (model) => async (req, res, next) => {
+    const page = parseInt(req.query.page);
+    const limit = parseInt(req.query.limit);
+
+    const results = {};
+    const userMap = {};
+    const startIndex = (page) * limit;
```

Comment:
Try to extract the common logic from the two pagination controllers (what pertains to page computation at least) to some single place/function.

Reply:
I think I solved that. I created a controller that handles the configuration of the pages. Please check configPage.js in the middleware folder.